doc/sphinx/source/glossary.rst

```diff
@@ -120,5 +120,42 @@ Glossary
    export
    
       The serialized version of a trained model that can be passed to
-      one of the nGraph backends for computation.      
+      one of the nGraph backends for computation.
 
+   NN
+
+      See :term:`ANN`.
+
+   ANN
+
+      :abbr:`ANN (Artificial Neural Network)` is an acronym for "Artificial 
+      Neural Network". ANN models are used to simulate possible combinations of 
+      binary logic processing and multi-layer (multi-dimensional) paths through 
+      which a :term:`data-flow graph` may be mapped or computed. An ANN does not 
+      have centralized storage; rather, an ANN manifests as information stored 
+      as patterns throughout the network structure. ANNs may be **Recurrent** 
+      (with feedback) or **Nonrecurrent** (feed-forward) with regard to the 
+      network structure. 
+
+   RNN 
+
+      See :term:`RANN`.
+
+
+   RANN 
+    
+      A :abbr:`Recurrent Artificial Neural Network (RANN)` is a variety of *ANN* 
+      where output nodes from a layer on a data-flow graph have loopback to 
+      nodes that created an earlier layer. Since the RANN has no "centralized" 
+      storage, this loopback is the means by which the ANN can "learn" or be 
+      trained.  There are several sub-categories of RANNs. An :term:`LSTM`, for 
+      example, is a specific kind of "recurrent" ANN with a particular way of 
+      implementing long/short-term memory. For brevity, the RANN acronym is 
+      often abbreviated as `RNN`.
+
+
+   LSTM
+
+      :abbr:`LSTM (Long Short-Term Memory)` is an acronym for "Long Short-Term 
```

Comment:
LSTMs add in mechanisms to keep chains of backprops from blowing up or going to 0, as they tend to do if you just use something simply for your cells.

Reply:
Something along the lines of: LSTMs extend on the traditional RNN by providing a number of ways to forget the memory of the previous time step via a set of learnable gates. These gates help avoid the problem of exploding or vanishing gradients that occur in the traditional RNN.